packages/core/upload/server/register.js

```diff
@@ -73,4 +74,9 @@ const baseProvider = {
   extend(obj) {
     Object.assign(this, obj);
   },
+  checkFileSize(file, { sizeLimit }) {
```

Comment:
we already get the entire config on createProvider. Any reason to pass it in this function again ?

Reply:
If I am not mistaken, inside the prover implementation you only have access to the providerOptions, but not the entire config.

For instance, in the [local provider ](https://github.com/strapi/strapi/pull/14398/files#diff-638964da9d2368ce7a6059512c81a6fd181ddb36f4c1589eae4f57904bb0680fR35)

Reply:
Yes you are right :) I'm was just talking about the base provider implementation here. but indeed that will keep them consistent.

Reply:
Oh, got it.
You are right, it should not be needed there. But yes! I would prefer the consistency of this.